src/yup.ts

```diff
@@ -38,41 +53,44 @@ const parseErrorSchema = (
         {},
       )
     : {
+        // @ts-expect-error
         [error.path]: { message: error.message, type: error.type },
       };
+};
+
+type ValidateOptions<T extends Yup.AnyObjectSchema> = Parameters<
+  T['validate']
+>[1];
 
-export const yupResolver = <TFieldValues extends FieldValues>(
-  schema: Yup.ObjectSchema | Yup.Lazy,
-  options: Omit<Yup.ValidateOptions, 'context'> = {
+export const yupResolver = <T extends Yup.AnyObjectSchema>(
+  schema: T,
```

Comment:
hi! 

i would like to hear about the reason why `Yup.Lazy` type schema is removed. 
from my understanding, from #31 we begun to support Lazy type schema but was removed in this PR. 

if it was meant to be, i would love hear about it. 
otherwise, i would make PR to fix it!

thanks~ @jorisre

Reply:
@sukjae 👋🏻 
It's a mistake

Feel free to send a PR ;)

Reply:
hi @jorisre 👋
nice! i made PR :) 
thanks~
#135